web/components/spluseins-calendar.vue

```diff
@@ -86,54 +169,73 @@ export default {
     }
   },
   methods: {
-    async next () {
-      // Jump forward 7 days, but do not refresh now (will be called in this.refresh())
-      this.calendar.next(7, true);
-      await this.refresh();
-    },
-    async prev () {
-      // Jump back 7 days
-      this.calendar.prev(7, true);
-      await this.refresh();
-    },
-    async today () {
-      this.resetWeek(true);
-      // This scrolls forward or backward until the today's week has been reached
-      while (this.calendar.start.date.isoWeek() > this.currentWeek) {
-        this.calendar.prev(7, true);
-      }
-      while (this.calendar.start.date.isoWeek() < this.currentWeek) {
-        this.calendar.next(7, true);
-      }
-      await this.refresh();
-    },
-    async refresh () {
-      // Store the week so switching calendars keeps the same week
-      this.setWeek(this.calendar.start.date.isoWeek());
-      // Load the data and store the events in the calendar
-      await this.load();
-      this.calendar.setEvents(this.events);
-      // Refresh calendar AFTER the events have been loaded to avoid flickering
-      this.calendar.refresh();
-    },
     ...mapMutations({
-      setWeek: 'splus/setWeek',
+      incrementWeek: 'splus/incrementWeek',
+      decrementWeek: 'splus/decrementWeek',
       resetWeek: 'splus/resetWeek'
     }),
     ...mapActions({
       load: 'splus/load'
-    })
+    }),
+    setToday () {
+      this.resetWeek(true);
+      this.load();
+    },
+    prev () {
+      this.decrementWeek();
+      this.load();
+    },
+    next () {
+      this.incrementWeek();
+      this.load();
```

Comment:
Hier war ein kleiner Trick drin — die alte Implementierung lädt erst die Daten und rendert dann neu. Dadurch gibt es beim Wechseln der Woche kein Flackern.
Das wäre nice to keep.

Reply:
Richtig, das hatte ich auch drin und hab's dann doch wieder rausgenommen (https://github.com/SplusEins/SplusEins/pull/474/commits/ab8253a11caa25a47beda7d05c41224eb24925be#diff-998e07fe03018e55d04823429d1d93389ff8c75243417f6a9d9eb377752e5e6aR59-R174). 

Das hat den Nachteil, dass man bei schlechten Verbindungen mobil dann nicht durch mehrere Wochen schnell nacheinander klicken kann, sondern immer abwarten muss, bis die neue Woche geladen ist.

Wir könnten es aber trotzdem in Kauf nehmen: Die Ladedauer ist nur bei sehr schlechten Verbindungen ("Slow 3G" throttling in Chrome) ungefähr 2 Sekunden. 

Da ich das Flackern aber auch nicht mag, werde ich das glaub ich dann wieder hinzufügen.